core/uploads-manager.php

```diff
@@ -0,0 +1,373 @@
+<?php
+namespace Elementor\Core;
+
+use Elementor\Core\Base\Base_Object;
+use Elementor\Core\Files\File_Types\Base as File_Type_Base;
+use Elementor\Core\Files\File_Types\Json;
+use Elementor\Core\Files\File_Types\Zip;
+use Elementor\Core\Utils\Exceptions;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Elementor uploads manager.
+ *
+ * Elementor uploads manager handler class is responsible for handling file uploads that are not done with WP Media.
+ *
+ * @since 3.2.0
+ */
+class Uploads_Manager extends Base_Object {
+
+	const UNFILTERED_FILE_UPLOADS_KEY = 'elementor_unfiltered_files_upload';
+	const INVALID_FILE_CONTENT = 'Invalid Content In File';
+
+	/**
+	 * @var File_Type_Base[]
+	 */
+	private $file_type_handlers = [];
+	private $allowed_mime_types;
+	private $are_unfiltered_files_enabled;
+
+	/**
+	 * Register File Types
+	 *
+	 * To Add a new file type to Elementor, with its own handling logic, you need to add it to the $file_types array here.
+	 *
+	 * @since 3.2.0
+	 */
+	public function register_file_types() {
+		// All file types that have handlers should be included here.
+		$file_types = [
+			'json' => new Json(),
+			'zip' => new Zip(),
+		];
+
+		foreach ( $file_types as $file_type => $file_handler ) {
+			$this->file_type_handlers[ $file_type ] = $file_handler;
+		}
+	}
+
+	/**
+	 * Extract and Validate Zip
+	 *
+	 * This method accepts a $file array (which minimally should include a 'tmp_name')
+	 *
+	 * @param array|null $file_path
+	 * @param array $allowed_file_types
+	 * @return array|\WP_Error
+	 */
+	public function extract_and_validate_zip( $file_path, $allowed_file_types = null ) {
+		$validation_result = [];
+
+		/** @var Zip $zip_handler - File Type */
+		$zip_handler = $this->file_type_handlers['zip'];
+
+		// Returns an array of file paths.
+		$extracted = $zip_handler->extract( $file_path, $allowed_file_types );
+
+		// If there are no extracted file names, no files passed the extraction validation.
+		if ( empty( $extracted['files'] ) ) {
+			// TODO: Decide what to do if no files passed the extraction validation
+			return new \WP_Error( 'file_error', self::INVALID_FILE_CONTENT );
+		}
+
+		foreach ( $extracted['files'] as $extracted_file ) {
+			// Each file is an array with a 'name' (file path) and 'type' (mime type) properties.
+			if ( $this->validate_file( $extracted_file ) ) {
+				$validation_result[] = $extracted_file;
+			}
+		}
+
+		return $validation_result;
```

Comment:
Return the same format as `$zip_handler->extract()`

Reply:
Done.